compiler/optimizer/VPHandlers.cpp

```diff
@@ -5704,6 +5704,13 @@ TR::Node *constrainAcall(TR::ValuePropagation *vp, TR::Node *node)
             TR::VPResolvedClass *newTypeConstraint = NULL;
             if (constraint)
                {
+               // Do nothing if the class of the object doesn't implement Cloneable
+               if (constraint->getClass() && !vp->comp()->fej9()->isCloneable(constraint->getClass()))
+                  {
+                  if (vp->trace())
+                     traceMsg(vp->comp(), "Object Clone: Class constraint on node %p doesn't exist or the class is not cloneable, quit\n", node);
+                  return node;
```

Comment:
That is a really good point - maybe we should add an unsuitable counter as well just so we know when we hit this case then?